nixos/modules/system/boot/stage-2-init.sh

```diff
@@ -71,22 +71,17 @@ fi
 
 
 # Provide a /etc/mtab.
-mkdir -m 0755 -p /etc
-test -e /etc/fstab || touch /etc/fstab # to shut up mount
+test -e /etc/fstab || install -D /dev/null /etc/fstab # to shut up mount
```

Comment:
this is bad IMO. Better be explicit in creating `/etc`, it isn't used *only* for `/etc/fstab`